docs/source/howto/codes.rst

```diff
@@ -4,41 +4,265 @@
+In other words, in order to use a |Parser| you will need to register it as explained in the following how-to section on :ref:`registering plugins <how-to:plugins>`.
+
+Handling parsing errors
+-----------------------
+
+In order for you to be able to provide the user with information regarding the errors that can ocur after the calculation has finished (so, mostly during parsing), you have the option of defining ``exit_codes`` for cases when something goes wrong.
+The typical way to implement these is to use ``try``/``except`` clauses to wrap the lines of code that might raise some typical errors, that can then be replaced by these ``exit_codes``:
+
+.. code-block:: python
+   :linenos:
+
+    def parse(self, **kwargs):
+        from aiida.common import exceptions
+
+        try:
+            output_folder = self.retrieved
+        except exceptions.NotExistent:
+            return self.exit_codes.ERROR_NO_RETRIEVED_FOLDER
+
+        output_filename = self.node.get_attribute('output_filename')
+
+        try:
+            with output_folder.open(output_filename, 'r') as handle:
+                try:
+                    result = int(handle.read())
+                except ValueError:
+                    result = None
+        except (OSError, IOError):
+            return self.exit_codes.ERROR_READING_OUTPUT_FILE
+
+        if result is None:
+            return self.exit_codes.ERROR_INVALID_OUTPUT
+
+        self.out('sum', orm.Int(result))
+
+You can see there is one for the case where no ``retrieve`` output was found (lines 4-7), another for the case of not being able to read the output file (lines 11-18), and finally one for when the result printed in the file is not a valid integer (lines 20-21).
+You then have to introduce all of these `exit_codes` inside of the |define| method of the |CalcJob| that will use this |Parser| (so, in that sense, when working like this the exit codes need to be "supported" by any calculation that wants to use the |Parser|).
+
+.. code-block:: python
+
+    @classmethod
+    def define(cls, spec):
+        (...)
+        spec.exit_code(300, 'ERROR_NO_RETRIEVED_FOLDER', message='The retrieved folder data node could not be accessed.')
+        spec.exit_code(310, 'ERROR_READING_OUTPUT_FILE', message='The output file could not be read from the retrieved folder.')
+        spec.exit_code(320, 'ERROR_INVALID_OUTPUT', message='The output file contains invalid output.')
+
+As you can see, for each ``exit_code`` you need to provide an ID number that will be used to identify it, a label you can then use to reference the code in the |parse| method (``self.exit_codes.LABEL``), and a message that will give the user more information on the problem.
+The Topics section on :ref:`defining processes <topics:processes:usage:defining>` also contains more information on how to use exit codes.
+
+Design guidelines
+-----------------
+
+ * | **Parse what you want to query for.**
+   | Make a list of which information to:
```

Comment:
This is a list with only one bullet point. Is there supposed to be another bullet point?

Reply:
The list of all guidelines was only one and I took out the point that was about parsing to put it in the parse section. I thought it would be better to keep the same formet even though that meant having a one item list, but now I tend to agree that its not so good. Anyways, if we go with your other suggestion I will just put all guidelines back together, otherwise I may remove this subsection and just add the content as a paragraph at the endo of the parse section.

Reply:
Let's keep it all together.